mobile/Screen/MapFilter.js

```diff
@@ -0,0 +1,206 @@
+import React, { useState } from 'react';
+import {
+  View, TouchableOpacity, ScrollView, Text, StyleSheet,
+} from 'react-native';
+import PropTypes from 'prop-types';
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    flexDirection: 'row',
+    flexWrap: 'wrap',
+    paddingHorizontal: 20,
+  },
+  text: {
+    fontSize: 16,
+    marginTop: 10,
+    marginLeft: 10,
+    color: 'grey',
+  },
+  header: {
+    color: 'dimgrey',
+    fontSize: 16,
+    marginTop: 25,
+    marginBottom: 10,
+    fontWeight: 'bold',
+    marginLeft: 10,
+  },
+  applyButton: {
+    marginTop: 20,
+    marginBottom: 50,
+    width: '95%',
+    borderRadius: 4,
+    height: 45,
+    backgroundColor: '#424288',
+    borderWidth: 2,
+    borderColor: '#424288',
+    justifyContent: 'center',
+    alignItems: 'center',
+    margin: 7,
+  },
+});
+
+function MapFilter({ route, navigation }) {
+  const {
+    categories, setCategories,
+  } = route.params;
+
+  const eventCategories = [
+    {
+      title: 'visual art',
+      length: 10,
+    },
+    {
+      title: 'film',
+      length: 4,
+    },
+    {
+      title: 'music',
+      length: 5,
+    },
+    {
+      title: 'art-related community event',
+      length: 27,
+    },
+    {
+      title: 'performance/theater',
+      length: 19,
+    },
+    {
+      title: 'spoken word',
+      length: 11,
+    },
+    {
+      title: 'miscellaneous',
+      length: 14,
+    },
+  ];
+
+  const resourceCategories = [
+    {
+      title: 'food',
+      length: 4,
+    },
+    {
+      title: 'shelter',
+      length: 7,
+    },
+    {
+      title: 'health',
+      length: 6,
+    },
+    {
+      title: 'legal services',
+      length: 14,
+    },
+    {
+      title: 'shower',
+      length: 6,
+    },
+    {
+      title: 'mission',
+      length: 7,
+    },
+    {
+      title: 'social services',
+      length: 15,
+    },
+  ];
+
+  const getButtonColor = (category) => {
+    if (category) {
+      return '#DFDFEF';
+    }
+    return '#F8F8F8';
+  };
+
+  const getBorderColor = (category) => {
+    if (category) {
+      return '#4C4C9B';
+    } return '#D0D0E8';
+  };
+
+  const getButtonStyle = (object, category) => ({
+    width: ((object.length) + 5) * 9,
```

Comment:
relating to my previous comment, this would just be `object.title.length`; that way we don't have to keep two fields

Reply:
and alternatively, if you just keep a list of strings, then it still be `object.length`, since `object` will now just be a string from the list. I recommend renaming the `object` variable as appropriate if you take this route